src/internal/core_actor.cc

```diff
@@ -860,8 +861,11 @@ caf::error core_actor_state::init_new_peer(endpoint_id peer,
   // message size. So on the wire, every byte block that our trait produces gets
   // prefixed with 32 bit with the size.
   namespace cn = caf::net;
-  auto [rd_1, wr_1] = caf::async::make_spsc_buffer_resource<node_message>();
-  auto [rd_2, wr_2] = caf::async::make_spsc_buffer_resource<node_message>();
+  // Note: structured bindings with values confuses clang-tidy's leak checker.
+  auto resources1 = caf::async::make_spsc_buffer_resource<node_message>();
+  auto& [rd_1, wr_1] = resources1;
```

Comment:
Same question re: references. The first two changes here make the variables references, but the third one doesn't.

Reply:
Good catch, the third one is an oversight. I'll fix it.